docs/DEVICE-CONNECTIVITY.md

```diff
@@ -1,142 +1,483 @@
 # Device Connectivity
 
-EVE needs to always be able to connect to the controller, yet the configuration of the network ports might be both complex and changing over time.
+EVE needs to always be able to connect to the controller, yet the configuration
+of the network ports might be both complex and changing over time.
+
+In addition, some network ports might be designated to application usage, or the underlying
+I/O bus controllers (e.g., PCI controllers) be designated for assignment to applications.
+That part of the configuration can also change in run-time.
+
+Any network port configuration changes which might affect the connectivity to the controller
+requires care to avoid permanently losing connectivity to the controller and become unmanageable
+as a result. Thus, it is required to perform testing as part of a configuration change
+and have a way to fall back to a working port configuration. This is accomplished by a logic
+to test connectivity to the controller, and persistently maintaining a list of network
+configurations, including the latest and possibly one or more fallback configurations.
+
+## Network adapters
+
+Network adapter denotes network configuration (MTU, MAC, IP, VLAN, etc.) and logical attributes
+(logical & interface names, cost, usage) attached to a physical network port.
+As show in the diagram below, there is a hierarchy of different types of network adapters
+and higher-level objects attached to them, starting with the physical ports on one side and
+ending with the EVE management plane and applications on the other side:
+
+![port-hierarchy](./images/eve-network-ports.png)
+
+Network adapters use logical labels for identification and for reference within the hierarchical
+layers. Unlike higher-level objects like apps or network instances that use UUIDs, these labels
+are user-defined, simpler human-readable strings, which are only required to be unique within
+the edge-node, not globally.
+
+### Physical network ports
+
+EVE supports ethernet NICs, WiFi modules and cellular modems.
+Support for [CAN devices is a work in-progress](https://zededa.atlassian.net/wiki/spaces/EVE/pages/2163539969/Introducing+CAN+support+on+EVE).
```

Comment:
This should be link to LF-Edge wiki, or?

Reply:
Yes, fixed.